src/main.rs

```diff
@@ -110,70 +113,66 @@ fn main() {
 	let password = create_rw_signal(String::from(""));
 	let error = create_rw_signal(String::from(""));
 	let timeout_que_id = create_rw_signal(0);
-	let config = Config::new();
+	let env = Environment::new(Config::new());
 	let que = Que::default();
 	let tooltip_signals = TooltipSignals::new(que);
 
-	let window_size = config.general.read().window_settings.window_size;
+	let window_size = env.config.general.read().window_settings.window_size;
 
 	let view = container(
 		dyn_container(
 			move || password.get(),
 			move |pass_value| {
+				let is_encrypted = env.db.config_db.read().encrypted;
+				if pass_value.is_empty() && !is_encrypted {
+					password.set(String::from(DEFAULT_DEBUG_PASSWORD)); // in debug mode - not encrypted and for debug only
+					return empty().any();
+				}
```

Comment:
I don't understand this bit? Before this was in the `if !is_unlocked && is_encrypted {} else {` block.

Reply:
It was double unlocking before. Changing the contents of password triggers the dynamic container function to run, so it would load the db and then load the db again.